10_code/40_WA_and_comparisons.py

```diff
@@ -0,0 +1,277 @@
+df_nv_county_yr = (
+    df_nv.groupby(["BUYER_STATE", "BUYER_COUNTY", "TRANSACTION_YEAR"])
+    .sum()
+    .reset_index()
+)
+df_nv_county_yr
+
+
+# # Idaho
+
+
+# Read the data for ID
+df_id = pd.read_csv(
+    "https://www.washingtonpost.com/wp-stat/dea-pain-pill-database/summary/arcos-id-statewide-itemized.csv.gz",
+    compression="gzip",
+)
+df_id.head()
+
+
+# filter columns of interest
+df_id = df_id[selected_col]
+df_id
+
+
+# assert no missing values in selected columns
+assert df_id["BUYER_STATE"].notnull().all()
+assert df_id["BUYER_COUNTY"].notnull().all()
+assert df_id["TRANSACTION_DATE"].notnull().all()
+assert df_id["MME"].notnull().all()
+
+
+df_id["TRANSACTION_DATE"] = pd.to_datetime(df_id["TRANSACTION_DATE"], format="%m%d%Y")
+df_id["TRANSACTION_YEAR"] = pd.DatetimeIndex(df_id["TRANSACTION_DATE"]).year
+# df_id["MONTH"] = pd.DatetimeIndex(df_id["TRANS_TIME"]).month
+df_id
+
+
+# Group by County and year and calculate opioid quantity.
+df_id_county_yr = (
+    df_id.groupby(["BUYER_STATE", "BUYER_COUNTY", "TRANSACTION_YEAR"])
+    .sum()
+    .reset_index()
+)
+df_id_county_yr
+
+
+# # Montana
+
+
+# Read the data for MT
+df_mt = pd.read_csv(
+    "https://www.washingtonpost.com/wp-stat/dea-pain-pill-database/summary/arcos-mt-statewide-itemized.csv.gz",
+    compression="gzip",
+)
+df_mt.head()
+
+
+# filter columns of interest
+df_mt = df_mt[selected_col]
+df_mt
+
+
+# assert no missing values in selected columns
+assert df_mt["BUYER_STATE"].notnull().all()
+assert df_mt["BUYER_COUNTY"].notnull().all()
+assert df_mt["TRANSACTION_DATE"].notnull().all()
+assert df_mt["MME"].notnull().all()
+
+
+df_mt["TRANSACTION_DATE"] = pd.to_datetime(df_mt["TRANSACTION_DATE"], format="%m%d%Y")
+df_mt["TRANSACTION_YEAR"] = pd.DatetimeIndex(df_mt["TRANSACTION_DATE"]).year
+# df_mt["MONTH"] = pd.DatetimeIndex(df_mt["TRANS_TIME"]).month
+df_mt
+
+
+# Group by County and year and calculate opioid quantity.
+df_mt_county_yr = (
+    df_mt.groupby(["BUYER_STATE", "BUYER_COUNTY", "TRANSACTION_YEAR"])
+    .sum()
+    .reset_index()
+)
+df_mt_county_yr
+
+
+# # Concatenate all the dataframes
+
+
+Opioid_Prescriptions_WA_comparisons = pd.concat(
+    (
+        df_wa_county_yr,
+        df_or_county_yr,
+        df_ca_county_yr,
+        df_nv_county_yr,
+        df_id_county_yr,
+        df_mt_county_yr,
+    ),
+    axis=0,
+)
+Opioid_Prescriptions_WA_comparisons
+
+
+# assert no missing values in selected columns
+assert Opioid_Prescriptions_WA_comparisons["BUYER_STATE"].notnull().all()
+assert Opioid_Prescriptions_WA_comparisons["BUYER_COUNTY"].notnull().all()
+assert Opioid_Prescriptions_WA_comparisons["TRANSACTION_YEAR"].notnull().all()
+assert Opioid_Prescriptions_WA_comparisons["MME"].notnull().all()
+
+
+# output to csv
+Opioid_Prescriptions_WA_comparisons.to_csv(
+    "https://github.com/MIDS-at-Duke/pds-2022-red-team/raw/main/20_intermediate_files/Opioid_Prescriptions_WA_comparisons.csv"
+)
```

Comment:
A few finishing thoughts: 
1. It looks like the current structure of the code is to (1) read from each csv file, then (2) process each file to remove null values and perform groupby, and finally (3) concatenate all processed files together. 
2. would it be helpful to define a function that performs tasks in (1) and (2)?